mteb/abstasks/statistics_calculation.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+from collections import Counter
+from typing import Any
+
+from mteb.types.statistics import (
+    ImageStatistics,
+    LabelStatistics,
+    ScoreStatistics,
+    TextStatistics,
+)
+
+
+def calculate_text_statistics(texts: list[str]) -> TextStatistics:
+    """Calculate descriptive statistics for a list of texts.
+
+    Args:
+        texts: List of texts to analyze.
+
+    Returns:
+        TextStatistics: A dictionary containing the descriptive statistics.
+    """
+    lengths = [len(text) for text in texts]
+    unique_texts = len(set(texts))
+
+    return TextStatistics(
+        total_text_length=sum(lengths),
+        min_text_length=min(lengths),
+        average_text_length=sum(lengths) / len(lengths),
+        max_text_length=max(lengths),
+        unique_texts=unique_texts,
+    )
+
+
+def calculate_image_statistics(images: list[Any]) -> ImageStatistics:
+    img_widths, img_heights = [], []
+    for img in images:
+        width, height = img.size  # type: ignore
+        img_heights.append(height)
+        img_widths.append(width)
+
+    return ImageStatistics(
+        min_image_width=min(img_widths),
+        average_image_width=sum(img_widths) / len(img_widths),
+        max_image_width=max(img_widths),
+        min_image_height=min(img_heights),
+        average_image_height=sum(img_heights) / len(img_heights),
+        max_image_height=max(img_heights),
+    )
```

Comment:
I am not sure if these are the statistics that people would normally be interested in?

Should we add duplicates as well?

Reply:
I would like duplicates - otherwise, I don't think I have a lot.

Reply:
What was previously in MIEB is fine